domain/keymanager/state/state.go

```diff
@@ -224,6 +225,55 @@ WHERE user_id = $userId.user_id
 		)
 	}
 
+	rval := make([]coressh.PublicKey, 0, len(publicKeys))
+	for _, pk := range publicKeys {
+		rval = append(rval, coressh.PublicKey{
+			Fingerprint: pk.Fingerprint,
+			Key:         pk.PublicKey,
+		})
+	}
+	return rval, nil
+}
+
+// GetPublicKeysDataForUser is responsible for returning all of the public keys
+// raw data for the user id in this model. If the user does not exist no error
+// is returned.
+func (s *State) GetPublicKeysDataForUser(
+	ctx context.Context,
+	id user.UUID,
+) ([]string, error) {
+	db, err := s.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	stmt, err := s.Prepare(`
+SELECT (public_key) AS (&publicKeyData.*)
+FROM user_public_ssh_key
+WHERE user_id = $userId.user_id
+`, userId{}, publicKeyData{})
+
+	if err != nil {
+		return nil, fmt.Errorf(
+			"preparing select statement for getting public keys data of user %q: %w",
+			id, domain.CoerceError(err),
+		)
+	}
+
+	userId := userId{id.String()}
+	publicKeys := []publicKeyData{}
```

Comment:
If you define these above you can pass them into `Prepare`. Only the type is important.

Reply:
They're different types. `[]foo` is not a `foo`. Unless I am missing something.

Reply:
You mean userId. Got it!